components/system-api/src/private.rs

```diff
@@ -206,59 +171,80 @@
-    pub id: InstanceId,
+/// Information about the current state of the node.
+#[derive(Debug, Clone, Copy, Serialize, Deserialize, PartialEq)]
+pub struct StatsInfo {
+    /// Height of the last committed block.
+    pub height: u64,
+    /// Total number of uncommitted transactions stored in persistent pool.
+    pub tx_pool_size: u64,
+    /// Total number of transactions in the blockchain.
+    pub tx_count: u64,
+    /// Size of the transaction cache.
+    pub tx_cache_size: usize,
+    /// Work duration of the node.
+    pub uptime: u64,
 }
 
-/// Short information about the current node.
-#[derive(Serialize, Deserialize, Clone, Debug, PartialEq)]
-pub struct NodeInfo {
-    /// Version of the `exonum` crate.
-    pub core_version: Option<String>,
+/// Information about whether it is possible to achieve the consensus between
+/// validators in the current state.
+#[derive(Deserialize, Serialize, Clone, Debug, PartialEq)]
+pub enum ConsensusStatus {
+    /// Consensus disabled on this node.
+    Disabled,
+    /// Consensus enabled on this node.
+    Enabled,
+    /// Consensus enabled and the node has enough connected peers.
+    Active,
 }
 
-impl NodeInfo {
-    /// Creates new `NodeInfo` from services list.
-    pub fn new() -> Self {
-        let core_version = option_env!("CARGO_PKG_VERSION").map(ToOwned::to_owned);
-        Self { core_version }
-    }
+/// Info about connected peer.
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq)]
+pub struct ConnectedPeerInfo {
+    /// Address of the peer.
+    pub address: String,
+    /// Consensus public key of the peer.
+    pub public_key: PublicKey,
+    /// Connect direction.
+    pub direction: String,
 }
 
-impl Default for NodeInfo {
-    fn default() -> Self {
-        Self::new()
+impl ConnectedPeerInfo {
+    fn new(connect_info: &ConnectInfo, direction: impl Into<String>) -> Self {
+        Self {
+            address: connect_info.address.to_owned(),
+            public_key: connect_info.public_key,
+            direction: direction.into(),
+        }
     }
 }
 
-/// Information about the outgoing connection.
-#[derive(Debug, Serialize, Deserialize, Default)]
-pub struct OutgoingConnection {
-    /// Public key of target node, if any.
-    pub public_key: Option<PublicKey>,
-}
-
-/// Information about connections of the node.
-#[derive(Debug, Serialize, Deserialize)]
-pub struct PeersInfo {
-    /// Incoming connections.
-    pub incoming_connections: Vec<ConnectInfo>,
-    /// Outgoing connections.
-    pub outgoing_connections: HashMap<SocketAddr, OutgoingConnection>,
+/// Short information about the current node.
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq)]
+pub struct NodeInfo {
+    /// Consensus status.
+    pub consensus_status: ConsensusStatus,
+    /// List of connected peers.
+    pub connected_peers: Vec<ConnectedPeerInfo>,
+    /// Version of the `exonum` crate.
+    pub exonum_version: String,
```

Comment:
Could this be made into a `semver::Version`?

Reply:
Potentially it could. But why ?